lix/packages/sdk/src/branch/get-leaf-changes-diff.ts

```diff
@@ -0,0 +1,113 @@
+import type { Change, LixReadonly } from "@lix-js/sdk";
+import type { LixPlugin } from "../plugin.js";
+
+export async function getLeafChangesDiff(args: {
+	plugins: LixPlugin[];
+	sourceDb: LixReadonly["db"];
+	targetDb?: LixReadonly["db"];
+	sourceBranchId?: string;
+	targetBranchId?: string;
+}): Promise<Change[]> {
+	if (!args.targetDb) {
+		args.targetDb = args.sourceDb;
+	}
+	if (!args.sourceBranchId) {
+		args.sourceBranchId = (
+			await args.sourceDb
+				.selectFrom("branch")
+				.select("id")
+				.where("active", "=", true)
+				.executeTakeFirstOrThrow()
+		).id;
+	}
+	if (!args.targetBranchId) {
+		args.targetBranchId = (
+			await args.targetDb
+				.selectFrom("branch")
+				.select("id")
+				.where("active", "=", true)
+				.executeTakeFirstOrThrow()
+		).id;
+	}
+
+	const leafChangesInSource = await args.sourceDb
+		.selectFrom("change_view")
+		.selectAll()
+		.where("branch_id", "=", args.sourceBranchId)
+		.where(
+			"id",
+			"not in",
+			// @ts-ignore - no idea what the type issue is
+			args.sourceDb
+				.selectFrom("change_view")
+				.select("parent_id")
+				.where("parent_id", "is not", undefined)
+				.where("branch_id", "=", args.sourceBranchId)
+				.distinct(),
+		)
+		.execute();
+
+	const sourceChangesByValueId = leafChangesInSource.reduce((agg, change) => {
+		// @ts-ignore
+		agg[change.value.id] = change;
+		return agg;
+	}, {});
+
+	const leafChangesInTarget = await args.targetDb
+		.selectFrom("change_view")
+		.selectAll()
+		.where("branch_id", "=", args.targetBranchId)
+		.where(
+			"id",
+			"not in",
+			// @ts-ignore - no idea what the type issue is
+			args.targetDb
+				.selectFrom("change_view")
+				.select("parent_id")
+				.where("parent_id", "is not", undefined)
+				.where("branch_id", "=", args.targetBranchId)
+				.distinct(),
+		)
+		.execute();
+
+	const targetChangesByValueId = leafChangesInTarget.reduce((agg, change) => {
+		// @ts-ignore
+		agg[change.value.id] = change;
+		return agg;
+	}, {});
+
+	const toDelete = [];
+	for (const change of leafChangesInTarget) {
+		// @ts-ignore
+		if (!sourceChangesByValueId[change.value.id]) {
+			// @ts-ignore
+			change._toDelete = true;
+			toDelete.push(change);
+		}
+	}
+
+	const plugin = args.plugins[0]!;
+	const toChange = [];
+
+	for (const change of leafChangesInSource) {
+		// @ts-ignore
+		const targetChange = targetChangesByValueId[change.value.id];
+		if (change.id === targetChange.id) {
+			continue;
+		}
+
+		// @ts-ignore
+		const old = targetChangesByValueId[change.value!.id] as Change | undefined;
+
+		const diff = await plugin.diff[change.type]!({
```

Comment:
I would check for the differ of the given type to exist and throw with an informative error like
```
 `diff function for change type ${change.type} not found in lix plugin`
```

Failing is way better than the silent skip - a problem i faced here so:
https://github.com/opral/monorepo/blob/lix-integration/lix/packages/sdk/src/file-handlers.ts#L143